File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/TimeoutReceiver.java
package jp.superwooo.chaipon.lighttimeswitcher;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class TimeoutReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String durationTypeName = intent.getStringExtra(AlarmScheduler.DurationTypeKey);
        Log.d("LS", "Time out receive: " + durationTypeName);
        DurationService durationService;
        try {
            durationService =
                    DurationType.valueOf(durationTypeName).create(context.getApplicationContext());
        }catch (Exception e){
            Log.e("LS", "duration type get error" + e.toString());
            durationService =
                    DurationType.Short.create(context.getApplicationContext());
        }
        durationService.setTimeOut();
        AlarmScheduler.scheduleAll(context.getApplicationContext());
    }
}

File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/TimeoutReceiver.kt
package jp.superwooo.chaipon.lighttimeswitcher

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import jp.superwooo.chaipon.lighttimeswitcher.AlarmScheduler.scheduleAll

class TimeoutReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val durationTypeName = intent.getStringExtra(AlarmScheduler.DurationTypeKey)
        Log.d("LS", "Time out receive: $durationTypeName")
        var durationService: DurationService?
        try {
            durationService =
                DurationType.valueOf(durationTypeName!!).create(context.applicationContext)
        } catch (e: Exception) {
            Log.e("LS", "duration type get error$e")
            durationService =
                DurationType.Short.create(context.applicationContext)
        }
        durationService!!.setTimeOut()
        scheduleAll(context.applicationContext)
    }
}
